Plugins: Install Grafana Pathfinder behind a feature flag

Jayclifford345 opened the pull request:
This pull request adds automatic installation for the new Pathfinder app behind a public preview feature toggle.

I've created [@grafana/pathfinder](https://github.com/orgs/grafana/teams/pathfinder/members) to be the CODEOWNERS of the feature. Do we need to give that team access to the repository in any way?

Jayclifford345 committed:
added grafana pathfinder behind a feature flag

Jayclifford345 committed:
fixed failing test

andresmgot commented on pkg/services/featuremgmt/registry.go:
```diff
@@ -1929,6 +1929,12 @@ var (
+		{
+			Name:        "grafanaPathfinder",
+			Description: "Enables Pathfinder app",
+			Stage:       FeatureStagePrivatePreview,
```
is this the correct stage? Usually new feature flags are added as `FeatureStageExperimental`

Jayclifford345 replied:
That should be it, updated @andresmgot

andresmgot reviewed (approved):
LGTM with a minor comment/question.

Jayclifford345 committed:
moved to experimental

Jayclifford345 committed:
chore: main into branch (#109953)

* Dashboard Schema V2: Handle a case when fieldConfig is undefined (#109881)

handle a case when fieldConfig is undefined

* Chore: Readd api client verification and update clients (#109847)

* add api client verification to lint step

* separate steps, clearer error messages

* make same change to non-enterprise version

* double-quote

* ignore conf

* update generated apis

* add names for steps

* fix comment

* Chore: regenerate api clients (#109919)

regenerate apis

* Chore: Remove duplicated `various-suite` cypress tests (#109737)

* remove duplicated various-suite cypress tests and update CODEOWNERS for playwright versions

* assign missing CODEOWNERS

* docs: fixing typo in config docs (#109882)

* MoveProvisionedDashboard: bug fix when selecting root folder, error is showing (#109638)

* MoveProvisionedDashboard: root folder error fix

* Remove provisioned badge when whole instance is provisioned, root folder checkbox fix

* If root level only have one provisioned folder, allow select all on subitems

* remove button tooltip and remove comment

* regenerate API clients after schema updates

---------

Co-authored-by: Alex Khomenko <Clarity-89@users.noreply.github.com>

* Provision resource operation: refresh parent folder when resource operation is performed (#109555)

* refresh parent folder when resource is created or deleted

* bulk delete and bulk move job: after job success, refetch folders

* refresh parent on both branch and default workflow

* move after success logic into getRepositoryJobsWithPath

* TableNG: Fix crash with malformed/invalid data links (#109920)

* CloudWatch: Fix logs query requestId to prevent setting undefined-logs as a requestId (#109930)

* Icon: Memoize rendered icons for situations where duplicate icons are rendered (#109883)

* CI: For community supported riscv64 builds use musl-cross instead of zig for… (#109931)

Community supported riscv64 builds: use musl-cross instead of zig for build

* DashboardPreviewBanner: isNewPr prop adjustment (#109885)

Bug fix where when saving changes to dashboard, DashboardPreviewBanner is rendering backward banner between loaded from new resource vs new resource created on remote.

* Transformation: Add missing binary operator (#109695)

* fix(transformation): add binary operator

* Implement a basic operator to reconcile the folder hierarchy from Unistore to Zanzana (#109705)

* Transformations: Fix `Partition by values` transform missing `refId` (#106417)

* Fix partition by values missing refId

* have partitioned refID include original refID

---------

Co-authored-by: Kristina Durivage <kristina.durivage@grafana.com>

* FS: Replace frontend-service-up with command that auto shuts down (#109933)

* Dashboard Migrations v16: Grid layout migration (#109603)

* Migration to be verified: v17 Convert minSpan to maxPerRow in panels

* Address review comments: use Go idiom for variable naming and sort.Ints for sorting

* Migration to be verified: 16 Grid layout migration

* Refactor v17 and v19 migrations to use shared helper functions

* Address reviews comments

---------

Co-authored-by: Haris Rozajac <58232930+harisrozajac@users.noreply.github.com>
Co-authored-by: Haris Rozajac <haris.rozajac12@gmail.com>

* Unified Storage: Add ListSinceModified to StorageBackend (#109697)

* WIP added ListSinceModified to StorageBackend interface

* fix compile time check

* Fix method name

* Fix naming

* fix the rest of the ListSinceModified names

* Uses resource key without name field

* get latest rv from resource_version. Update test.

* adds moar tests

* adds method stub for ListModifiedSince to other StorageBackend implementations

* adds dummy impl to noop storage backend for ListModifiedSince

* skip tests for badger kv backend for now

* fixes tests and adds badgerkv impl for ListModifiedSince

* add badger kv impl

* adds test for new query

* adds test data for new query

* adds ListModifiedSince stub to mockStorageBackend

* uncomment tests

* refactors ListModifiedSince to return an iter.seq2 and handles deduplication. Updates tests. Updates query result sorting.

* remove comments

* remove folder from query (dont need it, yet?)

* regen test queries

* updates test

* updates function comment

* use resourcepb.ResourceKey instead of ModifiedResourceKey

* wrap seq in single transaction. Rollback transaction after 30s if iterator never used. Only track last seen event. Formatting.

* skip TestListModifiedSince for kv backend

* use WatchEvent_Type for action type

* remove redundant fields from order by clause and regen test data for query

* remove redundant fields from order by clause and regen test data for query

* Migration v15: No-op migration for schema consistency (#109604)

* migrate to v19

* migrate to v18

* Migration to be verified: v17 Convert minSpan to maxPerRow in panels

* Migration to be verified: 16 Grid layout migration

* Refactor v17 and v19 migrations to use shared helper functions

* Migration to be verified: 15 No-op migration for schema consistency

* Update v27.go

* Update v16.go

* cleanup

---------

Co-authored-by: Haris Rozajac <haris.rozajac12@gmail.com>
Co-authored-by: Haris Rozajac <58232930+harisrozajac@users.noreply.github.com>

* Tooltip: Remove hideFrom.tooltip check for tooltip headers (#109925)

* Remove drone & dead code in pkg/build; update go modules (#109935)

* remove drone & dead code in pkg/build; update go modules

* remove .drone.star

* Remove drone scripts and drone references in Makefile

* make update-workspace

* remove deadcode tool

* Remove daggerbuild/scripts: deadcode

* Remove drone files / folders in CODEOWNERS

* make update-workspace

* remove more dead code

* E2E: Add performance monitoring E2E test (#109417)

* initial workflow

* add bench reporter

* add pr trigger

* update workflow

* actually add the test

* debug

.....

* more debug

* this it???

* Debugging GHA workflow

https

newline

tree

cache node modules

try bench again

fix path

* Add RequestsRecorder and tweak GHA workflow

* More debugging GHA workflow

cache

cache key

try renaming path

specify env var correctly

* Run tests on both envs, wait for requests to finish

fix status

cat stats

cat stats

* prefix metric names

* cleanup

* cleanup gha

* codeowners

* make gen-jsonnet

* quote things

* Add host_type label for categorising the various hosts

* rename pw_boot_time_seconds to pw_test_run_time_seconds

* change prefix

* codeowners, gha name

* remove node-modules caching

* fix codeowners

* Table: Refactor row height to work in pxs instead of lines (#109735)

* Table: Refactor row height to work in pxs instead of lines

* non-string pill value handling, internal cleanup on buildCellHeightMeasurers

* no need to round up to max line

* TableNG: Memoize some components to improve rendering performance (#109763)

* try React.memo

* Table: Rework the renderers file and some types

* more test cleanups

* remove for gauge for now

* memoize all of the renderers

* also memoize the Tooltip component

* memoize the TableCellActions

* memo useStyles2

* update useStyles2 usage in TableNG

* remove console.log

* add a test for the AutoCell fallback rendering case for testField

* style nit - update variable name to be more idiomatic

---------

Co-authored-by: Paul Marbach <paul.marbach@grafana.com>

* I18n: Download translations from Crowdin (#109946)

New Crowdin translations by GitHub Action

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

* Query Library: Rename "Query Library"to "Saved Queries" in Explore components (#109891)

* rename query library user interface text with "saved queries"

* add missing replacement on "Query history"

* fix unit test

* Fix test and apply PR feedback

* Frontend service: Fix favicon (#109914)

fix frontend service favicon

* Provisioning: Mark repository as unhealthy if hooks fail (#109788)

---------

Co-authored-by: Haris Rozajac <58232930+harisrozajac@users.noreply.github.com>
Co-authored-by: Ashley Harrison <ashley.harrison@grafana.com>
Co-authored-by: Jacob Valdez <jacob.valdez@grafana.com>
Co-authored-by: Yunwen Zheng <yunwen.zheng@grafana.com>
Co-authored-by: Alex Khomenko <Clarity-89@users.noreply.github.com>
Co-authored-by: Leon Sorokin <leeoniya@gmail.com>
Co-authored-by: Kevin Yu <kevinwcyu@users.noreply.github.com>
Co-authored-by: Paul Marbach <paul.marbach@grafana.com>
Co-authored-by: Kevin Minehart <5140827+kminehart@users.noreply.github.com>
Co-authored-by: Ihor Yeromin <yeryomin.igor@gmail.com>
Co-authored-by: Mihai Turdean <6640685+mihai-turdean@users.noreply.github.com>
Co-authored-by: KgOfHedgehogs <kgofhedgehogs@gmail.com>
Co-authored-by: Kristina Durivage <kristina.durivage@grafana.com>
Co-authored-by: Josh Hunt <joshhunt@users.noreply.github.com>
Co-authored-by: Dominik Prokop <dominik.prokop@grafana.com>
Co-authored-by: Haris Rozajac <haris.rozajac12@gmail.com>
Co-authored-by: owensmallwood <owen.smallwood@grafana.com>
Co-authored-by: Adela Almasan <88068998+adela-almasan@users.noreply.github.com>
Co-authored-by: grafana-pr-automation[bot] <140550294+grafana-pr-automation[bot]@users.noreply.github.com>
Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Alexa Vargas <239999+axelavargas@users.noreply.github.com>
Co-authored-by: Roberto Jiménez Sánchez <roberto.jimenez@grafana.com>

Jayclifford345 committed:
Merge https://github.com/grafana/grafana into default-grafana-pathfinder

Jayclifford345 commented:
Hey @tolzhabayev, will we hold off on merging this till the go no go meeting?

tolzhabayev commented:
Note from slack - probably targeting 12.3

jdbaldry committed:
Update feature toggle to public preview and update codeowners

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>

jdbaldry committed:
Merge remote-tracking branch 'origin/main' into default-grafana-pathfinder

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>
